### DISCUSSION TITLE
Differences between currentNote / originEntity / startNote (Backend API)

### DISCUSSION THREAD

**Discussion by Zocker1999NET** [TriliumNext/Trilium#4636] [Q&A] [answered]
From the [documentation](https://zadam.github.io/trilium/backend_api/BackendScriptApi.html), I cannot get the exact differences between `currentNote` / `originEntity` / `startNote`.

What I understand so far:
* in case of [entity events](https://github.com/zadam/trilium/wiki/Events#entity-events), `originEntity` describes the entity which triggered the execution of a script ([source](https://github.com/zadam/trilium/discussions/1856#discussioncomment-622305))
* otherwise, `currentNote` & `startNote` *seem* to be the note where the script is saved. However, I’m unsure because then one of them would be redundant …

**Comment by zerebos** [comment, marked as answer]
If I recall correctly, in the case of a script that has dependencies, when the sub-note-script dependencies get "required" their `startNote` is the parent script note while the `currentNote` is the note of the sub-script. But don't quote me on that it's been a while since I messed with those.

**Reply by zadam** [reply]
Yes, this is correct. `currentNote` is interesting only in case of scripts spread across multiple notes and even then you usually want to use `startNote`. I've improved the JSDoc a bit.
